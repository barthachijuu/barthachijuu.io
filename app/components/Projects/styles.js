/**
 *
 * Projects Styles
 *
 * @version 1.0.0
 * @author Bartolo Amico
 * @since 21-Jul-2020
 */

import { createUseStyles } from 'react-jss';

const projectsStyle = createUseStyles({

  projects: {
    background: '#ebeeee',
    paddingTop: '50px',
    paddingBottom: '60px',
    overflow: 'hidden',
    textAlign: 'left',

    '& h1': {
      fontSize: '21px',
      lineHeight: '24px',
      textTransform: 'uppercase',
      letterSpacing: '1px',
      textAlign: 'center',
      marginBottom: '48px',
      color: '#95a3a3',
    },
  },

  portfoliowrapper: {
    marginBottom: '36px',

    '& .columns': {
      marginBottom: '36px',
    },
  },

  itemwrap: {
    background: '#ffffff',
    overflow: 'hidden',
    position: 'relative',
    '-webkit-transition': 'all .3s ease-in-out',
    '-moz-transition': 'all .3s ease-in-out',
    '-o-transition': 'all .3s ease-in-out',
    '-ms-transition': 'all .3s ease-in-out',
    transition: 'all .3s ease-in-out',

    '& img': {
      verticalAlign: 'middle',
    },
  },

  overlay: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: '100%',
    height: '100%',
    opacity: 0,
    '-moz-opacity': 0,
    filter: 'alpha(opacity=0)',
    '-webkit-transition': 'opacity .3s ease-in-out',
    '-moz-transition': 'opacity .3s ease-in-out',
    '-o-transition': 'opacity .3s ease-in-out',
    transition: 'opacity .3s ease-in-out',
    background: 'url(images/overlay-bg.png) repeat',

    '&:hover': {
      opacity: 1,
      '-moz-opacity': 1,
      filter: 'alpha(opacity = 100)',
      cursor: 'pointer',
    },
  },

  itemmeta: {
    padding: '18px',

    '& h5': {
      fontSize: '15px',
      lineHeight: '24px',
      color: '#ffffff',
    },

    '& p': {
      fontSize: '12px',
      lineHeight: '18px',
      color: '#c6c7c7',
      marginBottom: 0,
    },
  },

  modalcontent: {
    maxWidth: '500px',
    maxHeight: '90%',
    overflow: 'auto',
    background: '#ffffff',
    position: 'fixed',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
  },

  description: {
    padding: '12px 36px 18px 36px',

    '& p': {
      fontSize: '14px',
      lineHeight: '24px',
      color: '#838c95',
      marginBottom: '12px',
    },
  },

  categories: {
    fontSize: '12px',
    lineHeight: '21px',
    color: '#838c95',
    textTransform: 'uppercase',
    letterSpacing: '2px',
    display: 'block',
    textAlign: 'left',

    '& i': {
      marginRight: '8px',
    },
  },

  link: {
    padding: '18px 36px',
    background: '#111111',
    textAlign: 'left',

    '& a, & button': {
      color: '#ffffff',
      fontSize: '11px',
      lineHeight: '21px',
      textTransform: 'uppercase',
      letterSpacing: '3px',
      cursor: 'pointer',
      outline: 'none',
      border: 0,
      background: 'transparent',
    },
  },

  popupdismiss: {
    marginLeft: '24px',
  },
});

export default projectsStyle;
