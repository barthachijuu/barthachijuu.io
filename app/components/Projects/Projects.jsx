/**
 *
 * Projects
 *
 * @version 1.0.0
 * @author Bartolo Amico
 * @since 21-Jul-2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'react-responsive-modal';
import styles from './styles.js';

const Projects = (props) => {
  const classes = styles();
  const getProjects = (project, index) => (
    <div className="columns">
      {/* eslint-disable */}
      <div className={classes.itemwrap} onClick={() => props.openModal({ status: true, idx: index })}>
      {/* eslint-enable */}
        <img
          src={project.image.thumb}
          alt={project.name}
        />
        <div className={classes.overlay}>
          <div className={classes.itemmeta}>
            <h5>{project.name}</h5>
            <p>{project.category}</p>
          </div>
        </div>
      </div>
      <Modal
        open={project.isOpen}
        onClose={() => props.closeModal({ status: false, idx: index })} /* eslint-disable-line */
        center
        classNames={{
          animationIn: 'customEnterAnimation',
          animationOut: 'customLeaveAnimation',
        }}
        animationDuration={1000}
      >
        <div className={classes.modalContent}>
          <img
            src={project.image.modal}
            alt={project.name}
          />
          <div className={classes.description}>
            <h5>{project.name}</h5>
            <p>{project.summary}</p>
            <span className={classes.categories}>
              <i className="fa fa-tag" />
              {project.keywords.join(', ')}
            </span>
          </div>
          <div className={classes.link}>
            <a
              href={project.website}
              target="_blank"
              rel="noopener noreferrer"
            >
              Details
            </a>
            <button
              type="button"
              className={classes.popupdismiss}
              onClick={() => props.closeModal({ status: false, idx: index })}
            >
              Close
            </button>
          </div>
        </div>
      </Modal>
    </div>
  );

  return (
    <>
      <section id="portfolio" className={classes.projects}>
        <div className="row">
          <div className="offset-2 ten columns collapsed">
            <h1>Portfolio</h1>
            <div id="portfolio-wrapper" className={`${classes.portfoliowrapper} bgrid-quarters s-bgrid-thirds cf`}>
              {props.projects.map((entry, index) => (
                getProjects(entry, index)
              ))}
            </div>
          </div>
        </div>
      </section>
    </>
  );
};

Projects.propTypes = {
  // Add here some propTypes
  // and set a minimal description i.e.
  /** Description of prop "projects". */
  projects: PropTypes.array.isRequired,
  /** Description of prop "openModal". */
  openModal: PropTypes.func.isRequired,
  /** Description of prop "closeModal". */
  closeModal: PropTypes.func.isRequired,
};

Projects.defaultProps = {
  // Add here some default propTypes values
};

export default Projects;
