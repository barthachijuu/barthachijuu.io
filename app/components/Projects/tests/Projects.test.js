/**
 *
 * Jest Unit Test for
 * Projects Component
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import Projects from '../Projects';

let baseProps;
let component;
let mockOpen;

beforeAll(() => {
  baseProps = {
    router: {
    },
  };
});

const projects = [
  {
    name: 'barthachijuu.io',
    publisher: 'barthachijuu',
    category: 'Web',
    releaseDate: '2016-09-01',
    website: 'https://github.com/barthachijuu/barthachijuu.io',
    summary: 'Lorem Ipsum',
    image: {
      modal: '/images/modal/barthachijuu.png',
      thumb: '/images/modal/thumb/barthachijuu_thumb.png',
    },
    keywords: [
      'Node.js',
      'React.js',
    ],
    isOpen: false,
  },
  {
    name: 'FE React Ark',
    publisher: 'barthachijuu',
    category: 'Web',
    releaseDate: '2019-04-14',
    website: 'https://github.com/barthachijuu/FeReactArk',
    summary: 'Lorem Ipsum Dolor Sit',
    image: {
      modal: '/images/modal/fereact.jpg',
      thumb: '/images/modal/thumb/fereact_thumb.jpg',
    },
    keywords: [
      'JavaScript',
      'React',
      'Node',
      'Webpack',
    ],
    isOpen: false,
  },
];

beforeEach(() => {
  mockOpen = jest.fn();
  component = shallow(<Projects {...baseProps} projects={projects} openModal={mockOpen} closeModal={mockOpen} />);
});

describe('<Projects />', () => {
  describe('Rendering', () => {
    it('should render self', () => {
      expect(component).toHaveLength(1);
    });
    it('test div click for openmodal', () => {
      component.find('#project0').simulate('click');
      expect(mockOpen.mock.calls.length).toEqual(1);
    });
    it('test button click for closemodal', () => {
      component.find('#dismiss0').simulate('click');
      expect(mockOpen.mock.calls.length).toEqual(1);
    });
    it('should match the snapshot', () => {
      expect(component.html()).toMatchSnapshot();
    });
  });
});
