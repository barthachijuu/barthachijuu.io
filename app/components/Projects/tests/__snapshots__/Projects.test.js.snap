// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> Rendering should match the snapshot 1`] = `"<section id=\\"portfolio\\" class=\\"projects-0-1-1\\"><div class=\\"row\\"><div class=\\"offset-2 ten columns collapsed\\"><h1>Portfolio</h1><div id=\\"portfolio-wrapper\\" class=\\"portfoliowrapper-0-1-2 bgrid-quarters s-bgrid-thirds cf\\"><div class=\\"columns\\"><div id=\\"project0\\" class=\\"itemwrap-0-1-3\\"><img src=\\"/images/modal/thumb/barthachijuu_thumb.png\\" alt=\\"barthachijuu.io\\"/><div class=\\"overlay-0-1-4\\"><div class=\\"itemmeta-0-1-5\\"><h5>barthachijuu.io</h5><p>Web</p></div></div></div></div><div class=\\"columns\\"><div id=\\"project1\\" class=\\"itemwrap-0-1-3\\"><img src=\\"/images/modal/thumb/fereact_thumb.jpg\\" alt=\\"FE React Ark\\"/><div class=\\"overlay-0-1-4\\"><div class=\\"itemmeta-0-1-5\\"><h5>FE React Ark</h5><p>Web</p></div></div></div></div></div></div></div></section>"`;
